acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Environments.scala

```diff
@@ -41,14 +41,49 @@ trait Environments {
     `start session and wait until it is ready`
     docsScreenshots.takeScreenshot()
 
-    projectPage.Sessions.Running.connectToJupyterLab
+    `connect to JupyterLab`
 
     Then("a JupyterLab page is opened on a new tab")
     val jupyterLabPage = JupyterLabPage()
     verify browserSwitchedTo jupyterLabPage sleep (5 seconds)
     jupyterLabPage
   }
 
+  private def `connect to JupyterLab`(implicit projectPage: ProjectPage): JupyterLabPage = eventually {
+    And("the user tries to connect to JupyterLab")
+    sleep(2 seconds)
+
+    if (projectPage.Sessions.Running.maybeOpenButton.fold(ifEmpty = true)(btn => !btn.isDisplayed))
+      pause whenUserCanSee { driver =>
+        val maybeGetLogs = projectPage.Sessions.Running.maybeOpenButton(driver)
+        if (maybeGetLogs.fold(false)(_.isDisplayed)) And("the 'Get logs' button is displayed")
+        maybeGetLogs
+      }
+
+    `try few times before giving up` { _ =>
+      And("clicks on the 'Open' button")
+      click on projectPage.Sessions.Running.openButtonMenu
+      sleep(2 seconds)
+
+      And("selects the 'Open in new tab' option")
+      click on projectPage.Sessions.Running.openInNewTabMenuItem
+      sleep(2 seconds)
+    }
+
+    if (webDriver.findTabWithUrlContaining("spawn-pending").nonEmpty) {
+      And("JupyterLab is not up yet")
+      Then("close the window and try again later")
+      // The server isn't up yet. Close the window and try again
```

Comment:
Is this comment necessary? I understand what's happening from the two lines above

Reply:
I don't know. I just moved this snippet from another place and actually removed quite a few other comments. Maybe this one actually explains something.

Reply:
Ha, ha. Actually, it's interesting why people tend to prefer commenting code over doing more effort while deciding on names.